packages/api/lib/writeRecords/write-granules.js

```diff
@@ -299,15 +373,33 @@ const _writeGranule = async ({
   log.info('About to write granule record %j to PostgreSQL', postgresGranuleRecord);
   log.info('About to write granule record %j to DynamoDB', dynamoGranuleRecord);
 
-  await createRejectableTransaction(knex, async (trx) => {
-    granuleCumulusId = await _writePostgresGranuleViaTransaction({
-      granuleRecord: postgresGranuleRecord,
-      executionCumulusId,
-      trx,
-      granulePgModel,
+  try {
+    await createRejectableTransaction(knex, async (trx) => {
+      granuleCumulusId = await _writePostgresGranuleViaTransaction({
+        granuleRecord: postgresGranuleRecord,
+        executionCumulusId,
+        trx,
+        granulePgModel,
+      });
+      return granuleModel.storeGranule(dynamoGranuleRecord);
     });
-    return granuleModel.storeGranule(dynamoGranuleRecord);
-  });
+  } catch (thrownError) {
+    log.error(`Write Granule failed: ${JSON.stringify(thrownError)}`);
+
+    if (isStatusFinalState(postgresGranuleRecord.status)
+      && isStatusFinalState(dynamoGranuleRecord.status)
```

Comment:
These should always be the same, so I'm inclined to say we just need `isStatusFinalState(dynamoGranuleRecord.status)` for now. Once Phase2 is delivered, it would switch to `isStatusFinalState(postgresGranuleRecord.status)`

Reply:
gotcha